### PR TITLE
ci: :bug: Fix security issue with checkout action and fix deploy actions to run when they should

### PR DESCRIPTION
**Pull request by lwjohnst86**
## Description

- This PR fixes the actions to only run when they should run (after the tests successfully pass and only on either the main or PRs). Also fixed some security issues with some actions (checkout to `v4`).

## Testing

- [ ] Yes
- [x] No, not needed (give a reason below)
- [ ] No, I need help writing them

Difficult to test actions.

## Reviewer Focus

This PR only needs a quick review.

### CODE REVIEW

File: common/actions/fly-deploy-pr.yml [hunk 1]
```diff
@@ -19,6 +19,9 @@ jobs:
   staging_app:
     runs-on: ubuntu-latest
 
+    # `workflow_run` doesn't work with `on.pull_request`, so need to use this `if` condition.
+    if: ${{ github.event.workflow_run.conclusion == 'success' && github.event.workflow_run.event == 'pull_request' }}
```
Comment (signekb) [inline common/actions/fly-deploy-pr.yml:23]: So, why is this needed? Not sure I understand the comment :)

File: common/actions/fly-deploy.yml [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
       - main
 
 jobs:
   deploy:
     name: Deploy app
     runs-on: ubuntu-latest
+
+    # Only run when test workflow passes.
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+
```
Comment (signekb) [inline common/actions/fly-deploy.yml:20]: Nice!

**Review by signekb — COMMENTED** [review]
Nice. Just have a question about one of the comments. Maybe it can be made a bit clearer? 🌻

**Comment by lwjohnst86** [comment]
@signekb is this clearer?

**Review by signekb — APPROVED** [review]
Ah! Yes, it's clearer now. Thanks!
